app/controllers/myott/mycommodities_controller.rb

```diff
@@ -1,6 +1,9 @@
 module Myott
   class MycommoditiesController < MyottController
-    before_action :authenticate, only: %i[new create index]
+    before_action :authenticate
+    before_action except: %i[new create] do
+      redirect_to myott_path unless current_subscription('my_commodities')
```

Comment:
Should this be
```suggestion
      redirect_to new_myott_mycommodity_path unless current_subscription('my_commodities')
```
I think it makes sense to take them to the upload page if they are in this controller without a subscription

Reply:
yes 👍